IM_calculation/IM/im_calculation.py

```diff
@@ -445,24 +508,58 @@ def compute_measures_multiprocess(
                     sorted(components_to_calculate, key=lambda x: x.value),
                 )
             )
-        all_results.extend(p.starmap(compute_measure_single, array_params))
-
-    all_result_dict = ChainMap(*all_results)
-    output_path = get_result_filepath(output, identifier, ".csv")
+            adv_array_params.append((waveform, advanced_im_config, output))
+        # only run simply im if and only if adv_im not going to run
+        if advanced_im_config.IM_list:
```

Comment:
As discussed in the last PR. Put this Adv IM block into a separate function.

Reply:
i've greatly trim the code and moved the logic to adv_im_factory.